Tasks/AzureFunctionOnKubernetesV0/src/utils/commandHelper.ts

```diff
@@ -0,0 +1,67 @@
+import * as path from "path";
+import * as tl from "azure-pipelines-task-lib/task";
+import * as tr from "azure-pipelines-task-lib/toolrunner";
+import * as FuncKubernetesUtility from 'kubernetes-common-v2/funckubernetesutility';
+import * as CommonUtils from 'kubernetes-common-v2/utility';
+import { DockerConnection } from "../dockerConnection";
+
+export class CommandHelper {
+    private funcPath: string;
+    public kubectlPath: string;
+
+    constructor() {
+        // Find the path of func and kubectl. This will throw if any of them is not present
+        // and hence the task will fail.
+        this.funcPath = tl.which('func', true);
+        this.kubectlPath = tl.which('kubectl', true);
+    }
+
+    public execCommand(command: tr.ToolRunner, options?: tr.IExecOptions, warnIfError?: boolean) {
+        const result: tr.IExecSyncResult = command.execSync(options);
+        CommonUtils.checkForErrors([result], warnIfError);
+        return result;
+    }
+    
+    public getFuncDeployCommand(dockerConnection: DockerConnection, secretName: string, appName: string, namespace: string, imageName: string, registry: string, pullSecretName: string, args: string): tr.ToolRunner {
+        if (!registry) {
+            registry = dockerConnection.getRegistry();
+        }
+
+        if (!imageName && !registry) {
+            tl.debug('Neither image-name nor registry input is provided. The deployment will fail.');
+        }
+
+        return FuncKubernetesUtility.getFuncDeployCommand(this.funcPath, secretName, appName, namespace, imageName, registry, pullSecretName, args);
+    }
+
+    public getCreateDockerRegistrySecretCommand(secretName: string, namespace: string): tr.ToolRunner {
+        const command = tl.tool(this.kubectlPath);
+        const dockerConfigDir = tl.getVariable('DOCKER_CONFIG');
```

Comment:
This could cause issues, scenario explained in a comment above

Reply:
Replied on above comment.